### PR TITLE
NFDIV-980 and NFDIV-1106 What you need to do notifications

### PR DESCRIPTION
**Pull request by lara-mpt**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/NFDIV-980
https://tools.hmcts.net/jira/browse/NFDIV-1106

### Change description ###
Modify What you need to do notifications for joint applicants

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/divorce/common/service/task/SendSubmissionNotifications.java [hunk 1]
```diff
@@ -59,8 +59,12 @@ private void sendCitizenNotifications(final CaseData caseData, final Long caseId
             }
         }
 
-        if (caseData.getApplication().hasAwaitingDocuments()) {
-            applicationOutstandingActionNotification.send(caseData, caseId);
+        if (caseData.getApplication().hasAwaitingApplicant1Documents()) {
+            applicationOutstandingActionNotification.sendToApplicant1(caseData, caseId);
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/divorce/common/service/task/SendSubmissionNotifications.java:63]: would be nice to add some logging here like app1 has awaiting docs so sending notification ...

File: src/main/java/uk/gov/hmcts/divorce/common/service/task/SendSubmissionNotifications.java [hunk 2]
```diff
@@ -59,8 +59,12 @@ private void sendCitizenNotifications(final CaseData caseData, final Long caseId
...
         }
+        if (caseData.getApplication().hasAwaitingApplicant2Documents()) {
+            applicationOutstandingActionNotification.sendToApplicant2(caseData, caseId);
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/divorce/common/service/task/SendSubmissionNotifications.java:66]: same here

File: src/test/java/uk/gov/hmcts/divorce/citizen/notification/ApplicationOutstandingActionNotificationTest.java [hunk 3]
```diff
@@ -81,10 +84,35 @@ void shouldCallSendEmailForSupportingDocuments() {
         verify(commonContent).templateVarsForApplicant(data, data.getApplicant1(), data.getApplicant2());
     }
 
+    @Test
+    @SuppressWarnings("squid:S6068")
```
Comment (nitinprabhu) [inline src/test/java/uk/gov/hmcts/divorce/citizen/notification/ApplicationOutstandingActionNotificationTest.java:88]: what is this suppression for ?

File: src/test/java/uk/gov/hmcts/divorce/citizen/notification/ApplicationOutstandingActionNotificationTest.java [hunk 4]
```diff
@@ -81,10 +84,35 @@ void shouldCallSendEmailForSupportingDocuments() {
...
+    void shouldCallSendEmailToApplicant2ForSupportingDocuments() {
+        CaseData data = validApplicant2CaseData();
+        data.getApplication().getMarriageDetails().setMarriedInUk(YesOrNo.NO);
+        data.getApplication().setApplicant2CannotUploadSupportingDocument(Set.of(NAME_CHANGE_EVIDENCE));
+
+        final HashMap<String, String> templateVars = new HashMap<>();
```
Comment (nitinprabhu) [inline src/test/java/uk/gov/hmcts/divorce/citizen/notification/ApplicationOutstandingActionNotificationTest.java:94]: can use Map.of(...) ?

File: src/test/java/uk/gov/hmcts/divorce/citizen/notification/ApplicationOutstandingActionNotificationTest.java [hunk 5]
```diff
@@ -81,10 +84,35 @@ void shouldCallSendEmailForSupportingDocuments() {
...
+        when(commonContent.templateVarsForApplicant(data, data.getApplicant2(), data.getApplicant1())).thenReturn(templateVars);
```
Comment (nitinprabhu) [inline src/test/java/uk/gov/hmcts/divorce/citizen/notification/ApplicationOutstandingActionNotificationTest.java:95]: do you want to return emptyMap() here ?

File: src/functionalTest/java/uk/gov/hmcts/divorce/citizen/CitizenAddPaymentFT.java [hunk 6]
```diff
@@ -94,4 +99,17 @@ public void shouldPassValidationAndGiveSuccessWhenCaseDataValidAndAwaitingDocume
...
 
+    @Test
+    public void shouldPassValidationAndGiveSuccessWhenCaseDataValidAndAwaitingDocumentJointApplication() throws IOException {
+        Map<String, Object> request = caseData(JOINT_AWAITING_DOCUMENTS_REQUEST);
+        Response response = triggerCallback(request, CITIZEN_ADD_PAYMENT, ABOUT_TO_SUBMIT_URL);
+
+        assertThat(response.getStatusCode()).isEqualTo(OK.value());
+
+        assertThatJson(response.asString())
+            .when(IGNORING_EXTRA_FIELDS)
```
Comment (nitinprabhu) [inline src/functionalTest/java/uk/gov/hmcts/divorce/citizen/CitizenAddPaymentFT.java:110]: can use TREATING_NULL_AS_ABSENT

File: src/test/java/uk/gov/hmcts/divorce/citizen/notification/ApplicationOutstandingActionNotificationTest.java (hunk 5 above)
Comment (lara-mpt) [reply src/test/java/uk/gov/hmcts/divorce/citizen/notification/ApplicationOutstandingActionNotificationTest.java:95]: @nitinprabhu yeah this is the same method most of the other notification unit tests have used to mock templateVarsForApplicant

File: src/test/java/uk/gov/hmcts/divorce/citizen/notification/ApplicationOutstandingActionNotificationTest.java (hunk 4 above)
Comment (lara-mpt) [reply src/test/java/uk/gov/hmcts/divorce/citizen/notification/ApplicationOutstandingActionNotificationTest.java:94]: @nitinprabhu same as below

File: src/test/java/uk/gov/hmcts/divorce/citizen/notification/ApplicationOutstandingActionNotificationTest.java (hunk 3 above)
Comment (lara-mpt) [reply src/test/java/uk/gov/hmcts/divorce/citizen/notification/ApplicationOutstandingActionNotificationTest.java:88]: not sure, I'll remove it. Just copied it in as every other test in the file has it

File: src/test/java/uk/gov/hmcts/divorce/citizen/notification/ApplicationOutstandingActionNotificationTest.java (hunk 5 above)
Comment (nitinprabhu) [reply src/test/java/uk/gov/hmcts/divorce/citizen/notification/ApplicationOutstandingActionNotificationTest.java:94]: what I meant was you can replace with below
```when(commonContent.templateVarsForApplicant(data, data.getApplicant1(), data.getApplicant2())).thenReturn(Collections.emptyMap());```

**Pull request closed without merging**